### PR TITLE
#997: Replace Optional.orElse() with Optional.orElseGet()

### LINKED ISSUE #997: Lazy evaluation of Optional fallback values
Hi!

I noticed that throughout the Citrus project, there are calls to `java.util.Optional#orElse()` with arguments that could be initialized or instantiated in a lazy manner, just to name a few:
- [InputAction](https://github.com/citrusframework/citrus/blob/main/core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java#L72)
- [KafkaProducer](https://github.com/citrusframework/citrus/blob/main/endpoints/citrus-kafka/src/main/java/org/citrusframework/kafka/endpoint/KafkaProducer.java#L75)

At those places, I think `Optional#orElseGet()` could be used instead.

I'll gladly open a PR, if you too think this could be useful.

### PR DESCRIPTION
**Pull request by picimako**
### Changes
- Replaced `Optional.orElse()` with `Optional.orElseGet()` where feasible, to evaluate non-static values lazily.

Fixes #997.

### CODE REVIEW

File: core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java [hunk 1]
```diff
@@ -69,7 +69,7 @@ public InputAction(Builder builder) {
         this.message = builder.message;
         this.variable = builder.variable;
         this.validAnswers = builder.validAnswers;
-        this.inputReader = Optional.ofNullable(builder.inputReader).orElse(new BufferedReader(new InputStreamReader(System.in)));
+        this.inputReader = Optional.ofNullable(builder.inputReader).orElseGet(() -> new BufferedReader(new InputStreamReader(System.in)));
```
Comment (bbortt) [inline core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java:72]: @christophd out of curiosity. how does citrus make sure this `inputReader` will be closed? I don't see it.

**Review by bbortt — APPROVED** [review]

File: core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java (hunk 1 above)
Comment (christophd) [reply core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java:72]: @bbortt I also do not see it being closed properly

Comment (bbortt) [reply core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java:72]: :octocat: @picimako would you be willing to take a look at this, too? please?

Comment (picimako) [reply core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java:72]: @bbortt Sure, I can look into it, and see what I can find.

Comment (bbortt) [reply core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java:72]: I think it's as easy as using a "try-with-resources" statement here:

```shell
diff --git a/core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java b/core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java
index 438310067..ea2409ee0 100644
--- a/core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java
+++ b/core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java
@@ -63,7 +63,7 @@ public class InputAction extends AbstractTestAction {
     /**
      * Default constructor.
      */
-    public InputAction(Builder builder) {
+    private InputAction(Builder builder) {
         super("input", builder);

         this.message = builder.message;
@@ -90,12 +90,9 @@ public class InputAction extends AbstractTestAction {
             display = message;
         }

-
-        try {
+        try (BufferedReader stdin = getInputReader()) {
             do {
                 logger.info(display);
-
-                BufferedReader stdin = getInputReader();
                 input = stdin.readLine();
             } while (validAnswers != null && !checkAnswer(input));
         } catch (IOException e) {
```

Comment (picimako) [reply core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java:72]: @bbortt Yeah, that should do it.

I also did a quick search, and I see that the same problem should be fixed in `org.citrusframework.validation.script.TemplateBasedScriptBuilder#build` too. If you have no objection, I'll open a separate PR for these two changes.

Comment (christophd) [reply core/citrus-base/src/main/java/org/citrusframework/actions/InputAction.java:72]: +1 on a separate PR for this. Many thanks!

**Review by bbortt — APPROVED** [review]

**Pull request merged**